### ISSUE TITLE
[EmptyLines]: End of Block not Marked as Empty but Acts as Empty.

### ISSUE THREAD

**Issue by darkwing** [devtools-html/debugger.html#4047]
Consider the following snippet:

```js
var TodoRouter = Backbone.Router.extend({
	routes: {
		'*filter': 'setFilter'
	},

	setFilter: function (param) {
		// Set the current filter to be used
		app.TodoFilter = param || '';

		// Trigger a collection filter event, causing hiding/unhiding
		// of Todo view items
		app.todos.trigger('filter');
	}
});

app.TodoRouter = new TodoRouter();
Backbone.history.start();
```

If I try to add a breakpoint at `});`, it shifts to the `app.TodoRouter` line.

We should include whatever logic shifts that breakpoint with the current EmptyLines  calculation.

**Comment by darkwing** [comment]
Of interest is the `getASTLocation` function in `src/components/astBreakpointLocation.js` -- that's where the shift happens.

**Comment by jasonLaster** [comment]
Here is a good sketch:

```diff
diff --git a/src/utils/parser/getEmptyLines.js b/src/utils/parser/getEmptyLines.js
index ef65820..d54d694 100644
--- a/src/utils/parser/getEmptyLines.js
+++ b/src/utils/parser/getEmptyLines.js
@@ -3,6 +3,7 @@ import difference from "lodash/difference";
 import { getAst } from "./utils/ast";

 const commentTokens = ["CommentBlock", "CommentLine"];
+const syntaxTokens = ["eof", "{", "}", "(", ")", ";"];

 function fillRange(start, end) {
   return Array(end - start + 1)
@@ -23,7 +24,8 @@ function getExecutableLines(ast) {
     .filter(
       token =>
         !commentTokens.includes(token.type) &&
-        (!token.type || (token.type.label && token.type.label != "eof"))
+        (!token.type ||
+          (token.type.label && !syntaxTokens.includes(token.type.label)))
     )
     .map(token => token.loc.start.line - 1);
```

we'll want to add this test to `getEmptyLines.js`  and change the test file to ``getEmptyLines.spec.js`

```js
it("outOfScope", () => {
    console.log(getEmptyLines(getSource("outOfScope")));
    expect(getEmptyLines(getSource("outOfScope"))).toMatchSnapshot();
  });
  ```

**Comment by jasonLaster** [comment]
I was thinking about these rules again and I think we want to add support for disabling these lines: (2, 3, 5, 6, 8, 10)

Here are the rules for disabling:

1. object key/value pairs where the value does not make a call
2. inert syntax like `}` and `});` on lines 5 and 10.
3. comments and whitespace (8, 6). 

```js
var TodoRouter = Backbone.Router.extend({          // 1
    routes: {                                      // 2
        '*filter': 'setFilter',                    // 3
        'key': doSomething()                       // 4
    },                                             // 5
                                                   // 6
    setFilter: function (param) {                  // 7
      // Set the current filter to be used         // 8
    }                                              // 9
});                                                // 10      
```

We can define inert syntax as a line with characters like `{}()`, that is not at the end of a function or beginning of a function. So for example, line 9 is the end of a function, but lines 5 and 10 are not.

**Comment by jasonLaster** [comment]
Daniel is on it

**Comment by jasonLaster** [comment]
We can close this as it will be addressed by the pause points work #5537
